public/create_new_email.php

```diff
@@ -0,0 +1,125 @@
+require_once "../modules/models/sent.php";
+?>
+
+<?php
+// Helper function to simply delete all objects contained in an array
+function delete_all($array)
+{
+    foreach ($array as $elem) {
+        $elem->delete();
+    }
+}
+
+if (isset($_POST["submit"])) {
+    // Get the logged in user
+    if (isset($_SESSION["current_user"])) {
+        $current_user = $_SESSION["current_user"];
+    } else {
+        echo "Error fetching logged in user.";
+        header("Location: login.php");
+    }
+
+    if (!empty($_POST["recipient_box"])) {
+        // Split "To" field by ; delimiter to obtain all receiving emails
+        $recipients = explode(";", $_POST["recipient_box"]);
+
+        // Validate each email.
+        $valid_emails = array();
+        $invalid_emails = array();
+        foreach ($recipients as $r) {
+            if (User::find_by_email($r) !== null) {
+                array_push($valid_emails, $r);
+            } else {
+                array_push($invalid_emails, $r);
+            }
+        }
+
+        // Create new email object
+        $email = new Email();
+        $email->subject = $_POST["subject_box"];
+        $email->content = $_POST["content"];
+
+        try {
+            $email->save();
+        } catch (PDOException $error) {
+            echo "<br>" . $error->getMessage();
+        }
+
+        // Add entry in inbox table for each user in "To" field
+        $saved_inboxes = array(); // There needs to be a better way to undo transactions. Maybe update the record class.
+        foreach ($valid_emails as $val) {
+            $inbox = new Inbox();
+            $inbox->email_address = $val;
+            $inbox->message_id = $email->id;
+
+            try {
+                $inbox->save();
+                array_push($saved_inboxes, $inbox);
+            } catch (PDOException $error) {
+                echo "<br>" . $error->getMessage();
+                $email->delete();
+                delete_all($saved_inboxes);
+            }
+        }
+
+        // Add entry in sent table for current_user->email
+        $sent = new Sent();
+        $sent->email_address = $current_user->email;
+        $sent->message_id = $email->id;
+
+        try {
+            $sent->save();
+        } catch (PDOException $error) {
+            echo "<br>" . $error->getMessage();
+            $email->delete();
+            delete_all($saved_inboxes);
+        }
+
+        // Tell user which emails could not be sent due to invalid addresses
+        if (!empty($invalid_emails)) {
+            // This can probably be done more elegantly
+            echo "<br>"."This message could not be sent to the following email addresses as they don't exist internally:"."<br>";
+            foreach ($invalid_emails as $inv) {
+                echo $inv . "\n";
+            }
+        }
+    } else { // If no recipients are specified
+        echo '<script>alert("Recipient field cannot be empty!")</script>';
+    }
```

Comment:
this alert thing feels sketchy, I think we are better of making the input field a required field
https://www.w3schools.com/tags/att_input_required.asp